### PR TITLE
[9.0][ADD] stock_picking_sale_order_link: Allow navigate from picking to sale order

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa

### CODE REVIEW

File: stock_picking_sale_order_link/README.rst [hunk 1]
```diff
@@ -0,0 +1,62 @@
+.. image:: https://img.shields.io/badge/licence-LGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/lgpl-3.0-standalone.html
+   :alt: License: LGPL-3
+
+=============================
+Stock Picking Sale Order Link
+=============================
+
+This module extends standard WMS to add a smart button in picking to go to
```
Comment (pedrobaeza) [inline stock_picking_sale_order_link/README.rst]: s/picking/pickings

File: stock_picking_sale_order_link/README.rst [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+sale order that creates this picking
```
Comment (pedrobaeza) [inline stock_picking_sale_order_link/README.rst]: put before `the`

File: stock_picking_sale_order_link/README.rst [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+
+Usage
+=====
+
+To use this module, you need to:
+
+#. Go to products and create one of type "Stockable".
+#. Got to sale order menu option and create one and confirm.
```
Comment (pedrobaeza) [inline stock_picking_sale_order_link/README.rst]: s/Got/Go

Comment (pedrobaeza) [reply stock_picking_sale_order_link/README.rst]: These are 2 steps at least. Unfold them.

File: stock_picking_sale_order_link/README.rst [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+#. Go to the picking and click in smartbutton.
```
Comment (pedrobaeza) [inline stock_picking_sale_order_link/README.rst]: Label of the smartbutton? Add the article `the` before `smartbutton`.

File: stock_picking_sale_order_link/models/sale_stock.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
```
Comment (pedrobaeza) [inline stock_picking_sale_order_link/models/sale_stock.py]: Name the file as `stock_picking.py`

File: stock_picking_sale_order_link/models/sale_stock.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+from openerp import api, models
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    @api.multi
+    def action_view_sale_order(self):
+        """This function returns an action that display existing sale orders
+        of given picking ids. It can either be a in a list or in a form view,
```
Comment (pedrobaeza) [inline stock_picking_sale_order_link/models/sale_stock.py]: This comment is not correct: you always return a form with one ID.

File: stock_picking_sale_order_link/models/sale_stock.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+        if there is only one delivery order to show.
+        """
+        self.ensure_one()
+        action = self.env.ref('sale.action_orders')
+        form = self.env.ref('sale.view_order_form', False)
+        form_id = form.id if form else False
```
Comment (pedrobaeza) [inline stock_picking_sale_order_link/models/sale_stock.py]: No need to do this in new API

File: stock_picking_sale_order_link/views/stock_picking_view.xml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    <record id="view_picking_form" model="ir.ui.view">
+        <field name="model">stock.picking</field>
+        <field name="inherit_id" ref="stock.view_picking_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//div[@name='button_box']" position="inside">
+                <field name="sale_id" invisible="1"/>
+                <button name="action_view_sale_order"
+                        class="oe_stat_button"
+                        icon="fa-dollar"
+                        string="Sale Order"
```
Comment (pedrobaeza) [inline stock_picking_sale_order_link/views/stock_picking_view.xml]: Sales Order

File: stock_picking_sale_order_link/README.rst (hunk 2 above)
Comment (pedrobaeza) [reply stock_picking_sale_order_link/README.rst]: s/sale/sales

File: stock_picking_sale_order_link/README.rst (hunk 3 above)
Comment (pedrobaeza) [reply stock_picking_sale_order_link/README.rst]: And put exact menu locations.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Review by luismontalba — APPROVED** [review]
Tested on runbot

**Review by chienandalu — APPROVED** [review]
- test reviewed

**Comment by sergio-teruel** [comment]
@pedrobaeza Changes done.

File: stock_picking_sale_order_link/README.rst [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+Usage
+=====
+
+To use this module, you need to:
+
+#. Go to *Sales > Products* and create one of type "Stockable".
+#. Go to *Sales > Sales Orders* and create one and confirm.
+#. Go to the picking generated clicking in delivery smartbutton.
+#. In the picking appears a new smartbutton to navegate to the sales order
```
Comment (sebalix) [inline stock_picking_sale_order_link/README.rst]: _to navigate_

**Review by sebalix — CHANGES_REQUESTED** [review]

**Comment by sergio-teruel** [comment]
@sebalix Changes done. Thanks!!

**Review by sebalix — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**